app/src/main/java/chat/rocket/android/authentication/infraestructure/SharedPreferencesMultiServerTokenRepository.kt

```diff
@@ -0,0 +1,33 @@
+package chat.rocket.android.authentication.infraestructure
+
+import chat.rocket.android.authentication.domain.model.TokenModel
+import chat.rocket.android.dagger.scope.PerActivity
+import chat.rocket.android.infrastructure.LocalRepository
+import chat.rocket.android.server.domain.MultiServerTokenRepository
+import com.squareup.moshi.Moshi
+
+@PerActivity
+class SharedPreferencesMultiServerTokenRepository(private val repository: LocalRepository,
+                                                  private val moshi: Moshi
+) : MultiServerTokenRepository {
+
+    override fun get(server: String): TokenModel? {
+        val token = repository.get("$TOKEN_KEY$server")
+        val adapter = moshi.adapter<TokenModel>(TokenModel::class.java)
+
+        token?.let {
```

Comment:
Forgot to say (in a past PR): thanks!
This is better than checking if the token isn't null before passing it in `.fromJson(token)`. And we can use the extension `ifNull`.
Thanks for the knowledge.
:clap: :clap: